larpandoracontent/LArHelpers/LArMCParticleHelper.cc

```diff
@@ -248,19 +248,47 @@ const MCParticle *LArMCParticleHelper::GetParentMCParticle(const MCParticle *con
 
 void LArMCParticleHelper::GetAllDescendentMCParticles(const pandora::MCParticle *const pMCParticle, pandora::MCParticleList &descendentMCParticleList)
 {
-    const MCParticleList &daughterMCParticleList = pMCParticle->GetDaughterList();
+    const MCParticleList &daughterMCParticleList(pMCParticle->GetDaughterList());
     for (const MCParticle *pDaughterMCParticle : daughterMCParticleList)
     {
         if (std::find(descendentMCParticleList.begin(), descendentMCParticleList.end(), pDaughterMCParticle) == descendentMCParticleList.end())
         {
-            descendentMCParticleList.push_back(pDaughterMCParticle);
+            descendentMCParticleList.emplace_back(pDaughterMCParticle);
             LArMCParticleHelper::GetAllDescendentMCParticles(pDaughterMCParticle, descendentMCParticleList);
         }
     }
 }
 
 //------------------------------------------------------------------------------------------------------------------------------------------
 
+void LArMCParticleHelper::GetAllDescendentMCParticles(const MCParticle *const pMCParticle, MCParticleList &descendentTrackParticles,
+    MCParticleList &leadingShowerParticles, MCParticleList &leadingNeutrons)
+{
+    const MCParticleList &daughterMCParticleList(pMCParticle->GetDaughterList());
```

Comment:
If going by Leigh's suggestion in LArPfoHelper.cc, can perhaps do the same here and remove this definition too...

Reply:
However I can see this appears in the function above as well...